docs/development/releasing.rst

```diff
@@ -1,233 +1,386 @@
-******************
-Release Procedures
-******************
+**********************************************
+Release procedure for the astropy core package
+**********************************************
 
-The current release procedure for Astropy involves a combination of an
-automated release script and some manual steps.  Future versions will automate
-more of the process, if not all.
+This page describes the release process for the core astropy package. For the average
+coordinated or affiliated package, you can instead check
+:ref:`these <simple-release-docs>` instructions which will be a lot simpler.
```

Comment:
It's tricky to do, because actually for most packages there are no release branches etc so it really does change this a lot - I think we just have to be better at keeping the other page up to date. I can go through it once this PR is finished.

Reply:
I still feel that those who end up doing releases are experienced enough to be able to not blindly copy-paste instructions but interpret whether something is branch-specific or not and therefore apply to their case or not. If some super basic release instruction ends up with the OA package template that's fine (and probably the right place), but I don't think there should be two versions maintained here, in this dev docs. In the past we did that approach, and they diverged and weren't really maintained super well. It's your call if you want to try that same approach again, I'm OK with either decision.

Reply:
I think it would make sense to include the simpler instructions in the OA guide, so I'll try and do that soon